### PR TITLE
[Feature] Add dishwasher app example

### PR DESCRIPTION
**Pull request by mideayanghui**
Problem:
Matter 1.2 clusters missing dishwasher app example

Solution:
Add dishwasher app example

### CODE REVIEW

**Comment by andy31415** [comment]
@mideayanghui please add this to build_examples.py build support (update `scripts/build/builders/host.py` and `scripts/build/build/targets.py`

**Review by woody-apple — APPROVED** [review]

File: examples/dishwasher-app/linux/include/CHIPProjectAppConfig.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+#pragma once
+
+// include the CHIPProjectConfig from config/standalone
+#include <CHIPProjectConfig.h>
+
+#define CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY 0
+
+// Bulbs do not typically use this - enabled so we can use shell to discover commissioners
+#define CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY_CLIENT 1
```
Comment (bzbarsky-apple) [inline examples/dishwasher-app/linux/include/CHIPProjectAppConfig.h:36]: What does this have to do with bulbs?

File: examples/dishwasher-app/linux/include/CHIPProjectAppConfig.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+#define CHIP_DEVICE_CONFIG_ENABLE_EXTENDED_DISCOVERY 1
```
Comment (bzbarsky-apple) [inline examples/dishwasher-app/linux/include/CHIPProjectAppConfig.h:38]: Why does this app need extended discovery?

File: examples/dishwasher-app/linux/include/CHIPProjectAppConfig.h [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+#define CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONABLE_DEVICE_TYPE 1
+
+#define CHIP_DEVICE_CONFIG_DEVICE_TYPE 257 // 0x0101 = 257 = Dimmable Bulb
```
Comment (bzbarsky-apple) [inline examples/dishwasher-app/linux/include/CHIPProjectAppConfig.h:42]: I don't think this is a bulb.

File: examples/dishwasher-app/linux/include/CHIPProjectAppConfig.h [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+#define CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONABLE_DEVICE_NAME 1
+
+#define CHIP_DEVICE_ENABLE_PORT_PARAMS 1
+
+#define CHIP_DEVICE_CONFIG_DEVICE_NAME "Test Bulb"
```
Comment (bzbarsky-apple) [inline examples/dishwasher-app/linux/include/CHIPProjectAppConfig.h:48]: This is not a bulb.

File: examples/dishwasher-app/linux/main.cpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+
+#if CHIP_DEVICE_CONFIG_ENABLE_WPA
+namespace {
+DeviceLayer::NetworkCommissioning::LinuxWiFiDriver sLinuxWiFiDriver;
+Clusters::NetworkCommissioning::Instance sWiFiNetworkCommissioningInstance(0, &sLinuxWiFiDriver);
+} // namespace
+#endif
```
Comment (bzbarsky-apple) [inline examples/dishwasher-app/linux/main.cpp:46]: This part is (1) wrong and (2) handled by AppInit now.  Please take out all the network commissioning bits and headers here.

File: examples/dishwasher-app/linux/main.cpp [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    {
+        return -1;
+    }
+
+#if defined(CHIP_IMGUI_ENABLED) && CHIP_IMGUI_ENABLED
+    example::Ui::ImguiUi ui;
+
+    ui.AddWindow(std::make_unique<example::Ui::Windows::QRCode>());
+    ui.AddWindow(std::make_unique<example::Ui::Windows::OccupancySensing>(chip::EndpointId(1), "Occupancy"));
+    ui.AddWindow(std::make_unique<example::Ui::Windows::Light>(chip::EndpointId(1)));
```
Comment (bzbarsky-apple) [inline examples/dishwasher-app/linux/main.cpp:69]: Does this dishwasher have occupancy sensing and a light on EP1?

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
